examples/model_service_template/mxnet_vision_batching.py

```diff
@@ -0,0 +1,219 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+import mxnet as mx
+import json
+import os
+import numpy as np
+from collections import namedtuple
+import logging
+
+
+class MXNetVisionServiceBatching(object):
+    def __init__(self):
+        """
+        Initialization for MXNet Vision Service supporting batch inference
+        """
+        self.mxnet_ctx = None
+        self.mx_model = None
+        self.labels = None
+        self.signature = None
+        self.epoch = 0
+        self.error = None
+        self._context = None
+        self._batch_size = 0
+        self.initialized = False
+        self.erroneous_reqs = set()
+
+    def top_probability(self, data, labels, top=5):
+        """
+        Get top probability prediction from NDArray.
+
+        :param data: NDArray
+            Data to be predicted
+        :param labels: List
+            List of class labels
+        :param top:
+        :return: List
+            List of probability: class pairs in sorted order
+        """
+        dim = len(data.shape)
+        if dim > 2:
+            data = mx.nd.array(
+                np.squeeze(data.asnumpy(), axis=tuple(range(dim)[2:])))
+        sorted_prob = mx.nd.argsort(data[0], is_ascend=False)
+        # pylint: disable=deprecated-lambda
+        top_prob = map(lambda x: int(x.asscalar()), sorted_prob[0:top])
+        return [{'probability': float(data[0, i].asscalar()), 'class': labels[i]}
+                for i in top_prob]
+
+    # noinspection PyMethodMayBeStatic
+    def get_model_files_prefix(self, context):
```

Comment:
Since we move this into single class, no need have a single line function any more.